pkg/resources/nifi/nifi.go

```diff
@@ -344,7 +344,19 @@ OUTERLOOP:
 			}
 
 			for _, volume := range node.Spec.Volumes {
-				if strings.HasPrefix(volume.Name, nifiutil.NifiDataVolumeMount) {
+				if volume.PersistentVolumeClaim == nil {
+					continue
+				}
+				pvcFound := &corev1.PersistentVolumeClaim{}
+				r.Client.Get(context.TODO(),
+					types.NamespacedName{
+						Name:      volume.PersistentVolumeClaim.ClaimName,
+						Namespace: r.NifiCluster.Namespace,
+					},
+					pvcFound,
+				)
+
+				if pvcFound.Labels[nifiutil.NifiDataVolumeMountKey] == "true" {
```

Comment:
If we're adding the label to every PVC attached to this pod (in `pvc.go`), why do we need to check if the PVC has the label here?

Reply:
Because you can have PVCs that are not attached to the pod using `StorageConfig` but using [ExternalVolumeConfigs](https://github.com/konpyutaika/nifikop/blob/034b1a2148e67cb9926d9dccfc78b7aae3ec6262/api/v1alpha1/nificluster_types.go#L291) and we don't want to delete them. 

And we check the value of the tag, because we could imagine in the future to add a parameter for `StoragConfig` to allow the destruction of non-conforming volume.

Reply:
I see. So then we only want to delete non-`ExternalVolumeConfigs` right? 

I understand why you've gone down this path, since determining the `StorageConfigs` for an individual `Node` can be challenging since [node configs for individual nodes and global are merged](https://github.com/konpyutaika/nifikop/blob/034b1a2148e67cb9926d9dccfc78b7aae3ec6262/pkg/util/util.go#L185-L200).

This makes sense. You never want to delete any `ExternalVolumeConfigs`